src/main/webapp/app/entities/programming-exercise/programming-exercise-instruction.component.ts

```diff
@@ -136,11 +136,9 @@ export class ProgrammingExerciseInstructionComponent implements OnChanges, OnDes
         if (this.resultSubscription) {
             this.resultSubscription.unsubscribe();
         }
-        return this.resultWebsocketService.subscribeResultForParticipation(this.participation.id).then(observable => {
-            this.resultSubscription = observable.pipe(distinctUntilChanged(({ id: id1 }: Result, { id: id2 }: Result) => id1 === id2)).subscribe(result => {
-                this.latestResult = result;
-                this.updateMarkdown();
-            });
+        return this.participationWebsocketService.subscribeForLatestResultOfParticipation(this.participation.id).subscribe(result => {
+            this.latestResult = result;
```

Comment:
Is it not necessary to create an unsubscribe variable here?

Reply:
you are right thanks. fixed that

Reply:
Cool!